crates/bitwarden/src/platform/fido2/types.rs

```diff
@@ -0,0 +1,256 @@
+use serde::Serialize;
+
+use super::{get_enum_from_string_name, SelectedCredential, Verification};
+
+#[cfg_attr(feature = "mobile", derive(uniffi::Record))]
+pub struct PublicKeyCredentialRpEntity {
+    pub id: String,
+    pub name: Option<String>,
+}
+
+#[cfg_attr(feature = "mobile", derive(uniffi::Record))]
+pub struct PublicKeyCredentialUserEntity {
+    pub id: Vec<u8>,
+    pub display_name: String,
+    pub name: String,
+}
+
+#[cfg_attr(feature = "mobile", derive(uniffi::Record))]
+pub struct PublicKeyCredentialParameters {
+    pub ty: String,
+    pub alg: i64,
+}
+
+impl TryFrom<PublicKeyCredentialParameters>
+    for passkey::types::webauthn::PublicKeyCredentialParameters
+{
+    type Error = crate::error::Error;
+
+    fn try_from(value: PublicKeyCredentialParameters) -> Result<Self, Self::Error> {
+        use coset::iana::EnumI64;
+        Ok(Self {
+            ty: get_enum_from_string_name(&value.ty)?,
+            alg: coset::iana::Algorithm::from_i64(value.alg).ok_or("Invalid algorithm")?,
+        })
+    }
+}
+
+#[cfg_attr(feature = "mobile", derive(uniffi::Record))]
+pub struct PublicKeyCredentialDescriptor {
+    pub ty: String,
+    pub id: Vec<u8>,
+}
+
+impl TryFrom<PublicKeyCredentialDescriptor>
+    for passkey::types::webauthn::PublicKeyCredentialDescriptor
+{
+    type Error = crate::error::Error;
+
+    fn try_from(value: PublicKeyCredentialDescriptor) -> Result<Self, Self::Error> {
+        Ok(Self {
+            ty: get_enum_from_string_name(&value.ty)?,
+            id: value.id.into(),
+            // TODO(Fido2): Do we need to expose this?
+            transports: None,
```

Comment:
As long as it is present in the response objects further down (which it is), no, not really. But it also doesn't hurt. Why? Is it not available here?

Reply:
Oh yeah I didn't notice the value was there, I'll add it